services/gateway/security/src/main/java/org/eclipse/ditto/services/gateway/security/authentication/jwt/PublicKeyProviderUnavailableException.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.services.gateway.security.authentication.jwt;
+
+import java.net.URI;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.model.base.common.HttpStatusCode;
+import org.eclipse.ditto.model.base.exceptions.DittoRuntimeException;
+import org.eclipse.ditto.model.base.exceptions.DittoRuntimeExceptionBuilder;
+import org.eclipse.ditto.model.base.headers.DittoHeaders;
+import org.eclipse.ditto.model.base.json.JsonParsableException;
+import org.eclipse.ditto.signals.commands.base.exceptions.GatewayException;
+
+@Immutable
+@JsonParsableException(errorCode = PublicKeyProviderUnavailableException.ERROR_CODE)
+public final class PublicKeyProviderUnavailableException extends DittoRuntimeException implements GatewayException {
+
+    public static final String ERROR_CODE = ERROR_CODE_PREFIX + "publickey.provider.unavailable";
+
+    private static final String DEFAULT_MESSAGE = "The public key provider is not available.";
+
+    private static final String DEFAULT_DESCRIPTION =
+            "If after retry it is still unavailable, please contact the service team.";
+
+    private static final long serialVersionUID = 1875913424069407158L;
+
+    private PublicKeyProviderUnavailableException(final DittoHeaders dittoHeaders,
+            @Nullable final String message,
+            @Nullable final String description,
+            @Nullable final Throwable cause,
+            @Nullable final URI href) {
+        super(ERROR_CODE, HttpStatusCode.SERVICE_UNAVAILABLE, dittoHeaders, message, description, cause, href);
+    }
+
+    /**
+     * A mutable builder for a {@code PublicKeyProviderUnavailableException}.
+     *
+     * @return the builder.
+     */
+    public static Builder newBuilder() {
```

Comment:
Since every call to the `PublicKeyProviderUnavailableException`  is `newBuilder().cause(xy).build()`, would it make sense to provide a factory method `withCause(throwable)` or at least a `newBuilder(throwable)`?

Reply:
I thought it'd be better to call `cause(xy)` rather than providing a single factory method `withCause(xy)`, because I would find it misleading. To me that would imply that a PublicKeyProviderUnavailableException can only be thrown with a cause, wich is not the case.